### PR TITLE
Use java test args for jvm based test rules

### PR DESCRIPTION
**Pull request by kageiit**

### CODE REVIEW

**Comment by kageiit** [comment]
cc @dreiss @styurin

**Comment by styurin** [comment]
build is broken

**Comment by kageiit** [comment]
Ant build works fine, but buck build is giving a cryptic error that looks like a jdk bug (https://bugs.openjdk.java.net/browse/JDK-8144101) apparently fixed in jdk 9. cc @jkeljo

File: src/com/facebook/buck/jvm/kotlin/KotlinTestDescription.java [hunk 1]
```diff
@@ -168,27 +164,5 @@ public void findDepsForTargetFromConstructorArgs(
...
-    default boolean getRunTestSeparately() {
-      return false;
-    }
-
-    @Value.Default
-    default ForkMode getForkMode() {
-      return ForkMode.NONE;
-    }
-  }
+      extends KotlinLibraryDescription.CoreArg, JavaTestDescription.CoreArg {}
```
Comment (styurin) [inline src/com/facebook/buck/jvm/kotlin/KotlinTestDescription.java]: by the way, you are bringing more by using JavaTestDescription.CoreArg, it inherits `JavaLibraryDescription.CoreArg`. The proper way is to create an abstract JvmBasedTestArg<T> where T is the type of library. You would be able to use it like this: 

    extends KotlinLibraryDescription.CoreArg, JvmBasedTestArg<KotlinLibraryDescription.CoreArg> {}

Comment (kageiit) [inline src/com/facebook/buck/jvm/kotlin/KotlinTestDescription.java]: `KotlinLibraryDescription.CoreArg` already inherits from `JavaLibraryDescription.CoreArg`. So this change doe snot bring in anything more

Comment (kageiit) [inline src/com/facebook/buck/jvm/kotlin/KotlinTestDescription.java]: @styurin any updates on this?

**Pull request closed without merging**